.github/workflows/test.yaml

```diff
@@ -125,12 +125,6 @@ jobs:
     - name: run integration tests
       run: make test.integration.dbless
 
-    - name: Upload coverage to Codecov
```

Comment:
How is this supposed to work? After this PR we'll have 3 jobs in parallel and only one of them uploading the test results to codecov. I'm not seeing any synchronization between these, so (if I'm not mistaken) we're relying on an implicit order of job finishes.

Also, how do we ensure that the coverage results from these 3 jobs (unit-tests, integration-tests-dbless, integrration-tests-postgres) get merged?

Please correct me if I misunderstood this change. But maybe we need:
- a barrier that will wait for all test runs producing coverage before we submit coverage to codecov
- a way of merging the coverage reports
?

Reply:
Ah, now I understand:
- the integration tests in the postgres flavor run with coverage
- the integration tests in the dbless flavor run without coverage
- the integration tests in both flavors run unit tests as well
- the unit tests don't run in isolation at all

I can see two problems with this approach that I'd like to make sure we address.
- there is an implicit expectation for the postgres target to produce coverage (which the dbless target does not do)
- we drop coverage for the dbless path which produces incorrect coverage for the dbless configuration path in KIC
- (soft problem introduced by a previous PR) the integration test target "silently" runs the unit tests as well

Actions required:
- implement a toggle for all test targets to enable/disable coverage explicitly
- ensure that CI uploads a coverage report from: dbless tests, integration tests, unit tests

Optional suggestion (because it goes beyond the context of this change):
- implement a shared test target for integration tests, with dbless/postgres implemented as a toggle
- de-duplicate dbless and postgres test jobs by using a parameter matrix (like we did for GKE tests in a previous PR)

Reply:
Given that this was just meant to be a small improvement to reduce noise and deduplicate coverage posts, and all the other things that need priority right now I'm going to opt to close this for now and we'll just have to put something into circulation to really holistically look at and fix our codecov later.